src/components/EarnReport.tsx

```diff
@@ -8,172 +13,480 @@ import Balance from './Balance'
 import Sprite from './Sprite'
 import styles from './EarnReport.module.css'
 
-interface YielgenReportTableProps {
-  lines: string[]
-  maxAmountOfRows?: number
+const SORT_KEYS = {
+  timestamp: 'TIMESTAMP',
+  cjTotalAmountInSats: 'CJ_TOTAL_AMOUNT_IN_SATS',
+  inputCount: 'INPUT_COUNT',
+  inputAmountInSats: 'INPUT_AMOUNT_IN_SATS',
+  feeInSats: 'FEE_IN_SATS',
+  earnedAmountInSats: 'EARNED_AMOUNT_IN_SATS',
 }
 
-const YieldgenReportTable = ({ lines, maxAmountOfRows = 15 }: YielgenReportTableProps) => {
+const TABLE_THEME = {
+  Table: `
+    --data-table-library_grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
+    font-size: 0.9rem;
+  `,
+  BaseCell: `
+    &:nth-of-type(2) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(3) button {
+      display: flex;
+      justify-content: center;
+    }
+    &:nth-of-type(4) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(5) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(6) button {
+      display: flex;
+      justify-content: end;
+    }
+  `,
+  Cell: `
+    &:nth-of-type(2) {
+      text-align: right;
+    }
+    &:nth-of-type(3) {
+      text-align: center;
+    }
+    &:nth-of-type(4) {
+      text-align: right;
+    }
+    &:nth-of-type(5) {
+      text-align: right;
+    }
+    &:nth-of-type(6) {
+      text-align: right;
+    }
+  `,
+}
+
+const TABLE_PAGE_SIZES = [25, 50, 100]
+
+type Minutes = number
+
+interface EarnReportEntry {
+  timestamp: Date
+  cjTotalAmount: Api.AmountSats | null
+  inputCount: number | null
+  inputAmount: Api.AmountSats | null
+  fee: Api.AmountSats | null
+  earnedAmount: Api.AmountSats | null
+  confirmationDuration: Minutes | null
+  notes: string | null
+}
+
+// in the form of yyyy/MM/dd HH:mm:ss - e.g 2009/01/03 02:54:42
+type RawYielgenTimestamp = string
+
+const parseYieldgenTimestamp = (val: RawYielgenTimestamp) => {
+  // adding the timezone manually will display the date in the users timezone correctly
+  return new Date(Date.parse(`${val} GMT`))
+}
+
+const yieldgenReportLineToEarnReportEntry = (line: string): EarnReportEntry | null => {
+  if (!line.includes(',')) return null
+
+  const values = line.split(',')
+
+  // be defensive here - we cannot handle lines with unexpected values
+  if (values.length < 8) return null
```

Comment:
Would `values.length > 8` be equally unexpected?

Reply:
Yes, equally unexpected, but having more values would not impair the component. Acceptable?

edit: It would have impacts if the order of the values change - but then again, that's also the case if a value is replaced, which would result in the same length. I will add some tests to ensure and document the data the component is expecting!

Reply:
Sounds good!

Reply:
Tests added!